src/MUIDataTable.js

```diff
@@ -202,7 +201,7 @@ class MUIDataTable extends React.Component {
     this.setHeadResizeable(this.headCellRefs, this.tableRef);
 
     // When we have a search, we must reset page to view it
-    if (this.props.options.searchText) this.setState({ page: 0 });
+    if (this.props.options.searchText && !this.props.options.page) this.setState({ page: 0 });
```

Comment:
Ok, how about `!this.props.options.serverside` instead of `!this.props.options.page`? Same for below. That way, we keep the old bugfix for people that are using custom pages, but we prevent `serverside` behavior from changing.

Reply:
That should work for us and keep the serverside behavior the same, but would it be confusing for anyone using it? I made the change so if we're good with it you can just accept it.